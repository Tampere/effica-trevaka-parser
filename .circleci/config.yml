# SPDX-FileCopyrightText: 2021 City of Tampere
#
# SPDX-License-Identifier: LGPL-2.1-or-later

version: 2.1

orbs:
  node: circleci/node@5.2.0

executors:
  reuse_tool:
    docker:
      - image: fsfe/reuse:0.13.0 # NOTE: Update bin/add-license-headers.sh to match
    resource_class: small
  machine:
    machine:
      image: ubuntu-2204:2024.01.1
    resource_class: medium
    environment:
      BUILDKIT_PROGRESS: plain

commands:
  checkout_with_submodules:
    steps:
      - checkout
      - run: git submodule sync
      - run: git submodule update --init --recursive

jobs:
  check_licenses:
    executor: reuse_tool
    steps:
      - checkout
      - run:
          name: reuse lint
          command: |
            reuse lint || {
              echo '';
              echo '=========================================';
              echo 'Please run: ./bin/add-license-headers.sh';
              echo '=========================================';
              exit 1
            }
  test:
    executor: machine
    steps:
      - checkout_with_submodules
      - run:
          name: Start up compose stack
          command: docker-compose -f docker-compose-it-db.yml up -d --wait --build
      - run:
          name: Waiting for compose stack to be up...
          command: docker-compose -f docker-compose-it-db.yml wait flyway
      - node/install-packages:
          include-branch-in-cache-key: false
      - run: npm run integration-test
      - run:
          name: Collect docker-compose logs
          command: docker-compose -f docker-compose-it-db.yml logs --tail=all > /tmp/docker-compose-logs.txt
          when: always
      - store_artifacts:
          path: /tmp/docker-compose-logs.txt
          destination: docker-compose-logs.txt

workflows:
  version: 2
  test:
    jobs:
      - check_licenses
      - test
