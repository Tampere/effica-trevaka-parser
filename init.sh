#!/bin/bash

# SPDX-FileCopyrightText: 2023-2024 Tampere region
#
# SPDX-License-Identifier: LGPL-2.1-or-later

set -e
set -u

function create_database() {
  psql -v ON_ERROR_STOP=1 --username "$POSTGRES_USER" <<-EOSQL
    CREATE DATABASE $1;
    \connect $1
    CREATE SCHEMA migration;
EOSQL
}

create_database vesilahti
