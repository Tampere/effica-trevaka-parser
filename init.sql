-- SPDX-FileCopyrightText: 2021 City of Tampere
--
-- SPDX-License-Identifier: LGPL-2.1-or-later

-- Add extensions here as needed (for UUIDs at least)
CREATE SCHEMA IF NOT EXISTS ext;
CREATE SCHEMA IF NOT EXISTS migration;

CREATE EXTENSION IF NOT EXISTS "uuid-ossp" WITH SCHEMA ext;
CREATE EXTENSION IF NOT EXISTS btree_gist WITH SCHEMA ext;