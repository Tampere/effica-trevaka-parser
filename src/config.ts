export const config = {
    cityVariant: process.env.CITY_VARIANT ?? "tampere",
    port: process.env.PORT ?? 3000,
    isTimed: process.env.ISTIMED?.toUpperCase() === "TRUE",
    migrationSchema: process.env.MIGRATION_SCHEMA ?? "migration",
    extensionSchema: process.env.EXTENSION_SCHEMA ?? "ext",
    migrationDb: {
        host: process.env.PGHOST ?? "localhost",
        port: 5432,
        user: process.env.PGUSER ?? "postgres",
        password: process.env.PGPASSWORD ?? "postgres",
        database: process.env.PGDATABASE ?? "migration",
    },
    xmlParserOptions: {
        parseNodeValue: true,
        arrayMode: true,
        trimValues: true,
        parseTrueNumberOnly: true
    },
    csvParserOptions: {
        trim: true,
        delimiter: "|"
    }
}