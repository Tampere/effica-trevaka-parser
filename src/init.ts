// SPDX-FileCopyrightText: 2021 City of Tampere
//
// SPDX-License-Identifier: LGPL-2.1-or-later

import migrationDb from "./db/db";
import { baseQueryParameters, runQueryFile } from "./util/queryTools";

export const initDb = async () => {
    return await migrationDb.tx(async (t) => {
        await runQueryFile("init.sql", t, baseQueryParameters);
        await runQueryFile("functions.sql", t, baseQueryParameters);
    });
};
