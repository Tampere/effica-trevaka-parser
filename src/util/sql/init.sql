-- SPDX-FileCopyrightText: 2021 City of Tampere
--
-- SPDX-License-Identifier: LGPL-2.1-or-later

CREATE TABLE IF NOT EXISTS ${migrationSchema:name}.idmap (
    type TEXT NOT NULL,
    effica_guid TEXT NOT NULL,
    evaka_id UUID NOT NULL,
    created TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT now(),
    updated TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT now(),
    PRIMARY KEY (type, effica_guid)
);
