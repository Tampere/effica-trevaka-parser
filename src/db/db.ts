import pgPromise from "pg-promise"
import { config } from "../config"

export const pgp = pgPromise({})

// date parsing seems to be scuffed, does local tz transformation but declares it UTC -> corrupts dates
//pgp.pg.types.setTypeParser(1082, dateParser)

const migrationDb = pgp({ ...config.migrationDb })

export default migrationDb