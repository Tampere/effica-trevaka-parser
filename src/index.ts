// SPDX-FileCopyrightText: 2021 City of Tampere
//
// SPDX-License-Identifier: LGPL-2.1-or-later

import app from "./app"
import { config } from "./config"

app.listen(config.port, () => {
    console.log(`Parser app listening at http://localhost:${config.port}`)
})

