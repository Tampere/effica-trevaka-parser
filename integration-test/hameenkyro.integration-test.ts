// SPDX-FileCopyrightText: 2023-2024 Tampere region
//
// SPDX-License-Identifier: LGPL-2.1-or-later

import { initDb } from "../src/init";
import { config as configFromEnv } from "../src/config";
import { pgp } from "../src/db/db";
import { cleanupDb } from "../src/util/testTools";
import {
    findPlacementMappings,
    findTransferPlacements,
    findTransformPlacements,
    imports,
    transfers,
    transforms,
    UNIT_COLUMN_SET,
} from "./test-utils";

const config = {
    ...configFromEnv,
    cityVariant: "hameenkyro",
    migrationDb: { ...configFromEnv.migrationDb, database: "hameenkyro" },
};
const db = pgp(config.migrationDb);
const importBasePath = `${__dirname}/data/sem/hameenkyro`;

beforeAll(() => db.tx(initDb));
beforeEach(() => db.tx(cleanupDb));
afterAll(db.$pool.end);

describe("hameenkyro", () => {
    beforeEach(async () => {
        await db.none(pgp.helpers.insert(units, UNIT_COLUMN_SET));
    });

    it("all placement mappings exists", async () => {
        const mappings = await db.tx((tx) =>
            findPlacementMappings(tx, config.cityVariant),
        );
        expect(mappings).toMatchSnapshot();
    });

    it("placement preschool", async () => {
        await imports(
            db,
            {
                path: `${importBasePath}/preschool-test/departments`,
                importTarget: "effica_department",
            },
            {
                path: `${importBasePath}/preschool-test/persons`,
                importTarget: "effica_person",
            },
            {
                path: `${importBasePath}/preschool-test/placements`,
                importTarget: "effica_placement",
            },
        );

        await transforms(db, config, "department", "person", "placement");
        expect(await db.tx(findTransformPlacements)).toMatchSnapshot();

        await transfers(db, "department", "person", "placement");
        expect(await db.tx(findTransferPlacements)).toMatchSnapshot();
    });

    it("placement preschool daycare", async () => {
        await imports(
            db,
            {
                path: `${importBasePath}/preschool-daycare-test/departments`,
                importTarget: "effica_department",
            },
            {
                path: `${importBasePath}/preschool-daycare-test/persons`,
                importTarget: "effica_person",
            },
            {
                path: `${importBasePath}/preschool-daycare-test/placements`,
                importTarget: "effica_placement",
            },
        );

        await transforms(db, config, "department", "person", "placement");
        expect(await db.tx(findTransformPlacements)).toMatchSnapshot();

        await transfers(db, "department", "person", "placement");
        expect(await db.tx(findTransferPlacements)).toMatchSnapshot();
    });

    it("placement private voucher service", async () => {
        await imports(
            db,
            {
                path: `${importBasePath}/voucher-placement-test/departments`,
                importTarget: "effica_department",
            },
            {
                path: `${importBasePath}/voucher-placement-test/persons`,
                importTarget: "effica_person",
            },
            {
                path: `${importBasePath}/voucher-placement-test/placements`,
                importTarget: "effica_placement",
            },
        );

        await transforms(db, config, "department", "person", "placement");
        expect(await db.tx(findTransformPlacements)).toMatchSnapshot();

        await transfers(db, "department", "person", "placement");
        expect(await db.tx(findTransferPlacements)).toMatchSnapshot();
    });
});

const units = [
    {
        id: "92b6e496-d273-11ee-9265-db72361ab90d",
        name: "Kunnallinen A",
        care_area_id: "b2402306-f415-4430-989c-16032490cc1b",
        provider_type: "MUNICIPAL",
    },
    {
        id: "ccf269c5-8677-452c-b2b7-e476a031f7b0",
        name: "Palveluseteli A",
        care_area_id: "b2402306-f415-4430-989c-16032490cc1b",
        provider_type: "PRIVATE_SERVICE_VOUCHER",
    },
];
