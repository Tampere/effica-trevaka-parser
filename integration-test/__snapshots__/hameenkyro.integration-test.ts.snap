// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hameenkyro all placement mappings exists 1`] = `
{
  "municipal": [
    {
      "effica": "*0-85 h/kk",
      "evaka": {
        "default_option": false,
        "id": "50358394-b961-11eb-b51f-67ac436e5637",
        "name_fi": "Varhaiskasvatus 85 tuntia / kuukausi",
        "valid_placement_type": "DAYCARE",
      },
    },
    {
      "effica": "*86-139 h/kk",
      "evaka": {
        "default_option": false,
        "id": "86ef70a0-bf85-11eb-91e6-1fb57a101165",
        "name_fi": "Varhaiskasvatus 86–120 tuntia / kuukausi",
        "valid_placement_type": "DAYCARE",
      },
    },
    {
      "effica": "* yli 140 h/kk",
      "evaka": {
        "default_option": false,
        "id": "503591ae-b961-11eb-b521-1fca99358eed",
        "name_fi": "Varhaiskasvatus yli 151 tuntia / kuukausi",
        "valid_placement_type": "DAYCARE",
      },
    },
    {
      "effica": "*Kokopäivähoito",
      "evaka": {
        "default_option": false,
        "id": "503591ae-b961-11eb-b521-1fca99358eed",
        "name_fi": "Varhaiskasvatus yli 151 tuntia / kuukausi",
        "valid_placement_type": "DAYCARE",
      },
    },
    {
      "effica": "Eo-täyd ph 0-85 h/k",
      "evaka": {
        "default_option": false,
        "id": "0a58d934-6fd1-11ed-a75e-c353faef5858",
        "name_fi": "Täydentävä varhaiskasvatus 0–50 tuntia / kuukausi",
        "valid_placement_type": "PRESCHOOL_DAYCARE_ONLY",
      },
    },
    {
      "effica": "Esiopetus 4t/päivä",
      "evaka": {
        "default_option": true,
        "id": "94e44ef1-106b-401d-81b6-8e5c31cd0437",
        "name_fi": "Esiopetus",
        "valid_placement_type": "PRESCHOOL",
      },
    },
    {
      "effica": "Vuorohoito 0-85 h/kk",
      "evaka": {
        "default_option": false,
        "id": "50358394-b961-11eb-b51f-67ac436e5637",
        "name_fi": "Varhaiskasvatus 85 tuntia / kuukausi",
        "valid_placement_type": "DAYCARE",
      },
    },
    {
      "effica": "Tilapäinen hoito, kp",
      "evaka": {
        "default_option": true,
        "id": "e1063bee-c19d-469d-85a5-6b0350872d76",
        "name_fi": "Tilapäinen varhaiskasvatus",
        "valid_placement_type": "TEMPORARY_DAYCARE",
      },
    },
    {
      "effica": "Vuorohoito 86-120 h/kk",
      "evaka": {
        "default_option": false,
        "id": "86ef70a0-bf85-11eb-91e6-1fb57a101165",
        "name_fi": "Varhaiskasvatus 86–120 tuntia / kuukausi",
        "valid_placement_type": "DAYCARE",
      },
    },
    {
      "effica": "*Eo-täyd. osa-aikainen",
      "evaka": {
        "default_option": false,
        "id": "0a58db0a-6fd1-11ed-a75e-bbde95c1aded",
        "name_fi": "Täydentävä varhaiskasvatus 51–85 tuntia / kuukausi",
        "valid_placement_type": "PRESCHOOL_DAYCARE_ONLY",
      },
    },
    {
      "effica": "Eo-täyd ph 86-120 h/kk",
      "evaka": {
        "default_option": false,
        "id": "0a58dbe6-6fd1-11ed-a75e-5335f2b9a91c",
        "name_fi": "Täydentävä varhaiskasvatus 86–120 tuntia / kuukausi",
        "valid_placement_type": "PRESCHOOL_DAYCARE_ONLY",
      },
    },
    {
      "effica": "Pkoti/ aikap. 0-85 h/kk",
      "evaka": {
        "default_option": false,
        "id": "50358394-b961-11eb-b51f-67ac436e5637",
        "name_fi": "Varhaiskasvatus 85 tuntia / kuukausi",
        "valid_placement_type": "DAYCARE",
      },
    },
    {
      "effica": "Vuorohoito 121-140 h/kk",
      "evaka": {
        "default_option": false,
        "id": "503590f0-b961-11eb-b520-53740af3f7ef",
        "name_fi": "Varhaiskasvatus 121-150 tuntia / kuukausi",
        "valid_placement_type": "DAYCARE",
      },
    },
    {
      "effica": "Vuorohoito 141-155 h/kk",
      "evaka": {
        "default_option": false,
        "id": "503590f0-b961-11eb-b520-53740af3f7ef",
        "name_fi": "Varhaiskasvatus 121-150 tuntia / kuukausi",
        "valid_placement_type": "DAYCARE",
      },
    },
    {
      "effica": "Vuorohoito yli 171 h/kk",
      "evaka": {
        "default_option": false,
        "id": "503591ae-b961-11eb-b521-1fca99358eed",
        "name_fi": "Varhaiskasvatus yli 151 tuntia / kuukausi",
        "valid_placement_type": "DAYCARE",
      },
    },
    {
      "effica": "Pph/ aikap. 141-155 h/kk",
      "evaka": {
        "default_option": false,
        "id": "503590f0-b961-11eb-b520-53740af3f7ef",
        "name_fi": "Varhaiskasvatus 121-150 tuntia / kuukausi",
        "valid_placement_type": "DAYCARE",
      },
    },
    {
      "effica": "Pph/ aikap. 156-170 h/kk",
      "evaka": {
        "default_option": false,
        "id": "503591ae-b961-11eb-b521-1fca99358eed",
        "name_fi": "Varhaiskasvatus yli 151 tuntia / kuukausi",
        "valid_placement_type": "DAYCARE",
      },
    },
    {
      "effica": "Pph/ aikap. yli 171 h/kk",
      "evaka": {
        "default_option": false,
        "id": "503591ae-b961-11eb-b521-1fca99358eed",
        "name_fi": "Varhaiskasvatus yli 151 tuntia / kuukausi",
        "valid_placement_type": "DAYCARE",
      },
    },
    {
      "effica": "Pkoti/ aikap. 86-120 h/kk",
      "evaka": {
        "default_option": false,
        "id": "86ef70a0-bf85-11eb-91e6-1fb57a101165",
        "name_fi": "Varhaiskasvatus 86–120 tuntia / kuukausi",
        "valid_placement_type": "DAYCARE",
      },
    },
    {
      "effica": "*Kokopäivähoito 12 pv/kk",
      "evaka": {
        "default_option": false,
        "id": "50358394-b961-11eb-b51f-67ac436e5637",
        "name_fi": "Varhaiskasvatus 85 tuntia / kuukausi",
        "valid_placement_type": "DAYCARE",
      },
    },
    {
      "effica": "Pkoti/ aikap. 121-140 h/kk",
      "evaka": {
        "default_option": false,
        "id": "503590f0-b961-11eb-b520-53740af3f7ef",
        "name_fi": "Varhaiskasvatus 121-150 tuntia / kuukausi",
        "valid_placement_type": "DAYCARE",
      },
    },
    {
      "effica": "Pkoti/ aikap. 141-155 h/kk",
      "evaka": {
        "default_option": false,
        "id": "503590f0-b961-11eb-b520-53740af3f7ef",
        "name_fi": "Varhaiskasvatus 121-150 tuntia / kuukausi",
        "valid_placement_type": "DAYCARE",
      },
    },
    {
      "effica": "Pkoti/ aikap. 156-170 h/kk",
      "evaka": {
        "default_option": false,
        "id": "503591ae-b961-11eb-b521-1fca99358eed",
        "name_fi": "Varhaiskasvatus yli 151 tuntia / kuukausi",
        "valid_placement_type": "DAYCARE",
      },
    },
    {
      "effica": "Pkoti/ aikap. yli 171 h/kk",
      "evaka": {
        "default_option": false,
        "id": "503591ae-b961-11eb-b521-1fca99358eed",
        "name_fi": "Varhaiskasvatus yli 151 tuntia / kuukausi",
        "valid_placement_type": "DAYCARE",
      },
    },
  ],
  "voucher": [
    {
      "effica": "*0-85 h/kk",
      "evaka": {
        "default_option": false,
        "id": "36c8c2ed-7543-47de-bc42-14d163a6277d",
        "name_fi": "Palveluseteli Varhaiskasvatus 85 tuntia / kuukausi",
      },
    },
    {
      "effica": "*86-139 h/kk",
      "evaka": {
        "default_option": false,
        "id": "9eb82822-82ef-46fb-a123-29cf0af757b7",
        "name_fi": "Palveluseteli Varhaiskasvatus 86–139 tuntia / kuukausi",
      },
    },
    {
      "effica": "* yli 140 h/kk",
      "evaka": {
        "default_option": false,
        "id": "f5d32585-2c78-4434-95d6-30b446db7d4d",
        "name_fi": "Palveluseteli Varhaiskasvatus yli 140 tuntia / kuukausi",
      },
    },
    {
      "effica": "*Kokopäivähoito",
      "evaka": {
        "default_option": false,
        "id": "f5d32585-2c78-4434-95d6-30b446db7d4d",
        "name_fi": "Palveluseteli Varhaiskasvatus yli 140 tuntia / kuukausi",
      },
    },
    {
      "effica": "Eo-täyd ph 0-85 h/k",
      "evaka": {
        "default_option": false,
        "id": "fe0972a5-6ce9-41cc-a635-82fb22e7891b",
        "name_fi": "Palveluseteli Täydentävä varhaiskasvatus",
      },
    },
    {
      "effica": "Esiopetus 4t/päivä",
      "evaka": {
        "default_option": true,
        "id": "94e44ef1-106b-401d-81b6-8e5c31cd0437",
        "name_fi": "Esiopetus",
      },
    },
    {
      "effica": "Vuorohoito 0-85 h/kk",
      "evaka": {
        "default_option": false,
        "id": "36c8c2ed-7543-47de-bc42-14d163a6277d",
        "name_fi": "Palveluseteli Varhaiskasvatus 85 tuntia / kuukausi",
      },
    },
    {
      "effica": "Tilapäinen hoito, kp",
      "evaka": {
        "default_option": true,
        "id": "e1063bee-c19d-469d-85a5-6b0350872d76",
        "name_fi": "Tilapäinen varhaiskasvatus",
      },
    },
    {
      "effica": "Vuorohoito 86-120 h/kk",
      "evaka": {
        "default_option": false,
        "id": "9eb82822-82ef-46fb-a123-29cf0af757b7",
        "name_fi": "Palveluseteli Varhaiskasvatus 86–139 tuntia / kuukausi",
      },
    },
    {
      "effica": "*Eo-täyd. osa-aikainen",
      "evaka": {
        "default_option": false,
        "id": "fe0972a5-6ce9-41cc-a635-82fb22e7891b",
        "name_fi": "Palveluseteli Täydentävä varhaiskasvatus",
      },
    },
    {
      "effica": "Eo-täyd ph 86-120 h/kk",
      "evaka": {
        "default_option": false,
        "id": "fe0972a5-6ce9-41cc-a635-82fb22e7891b",
        "name_fi": "Palveluseteli Täydentävä varhaiskasvatus",
      },
    },
    {
      "effica": "Pkoti/ aikap. 0-85 h/kk",
      "evaka": {
        "default_option": false,
        "id": "36c8c2ed-7543-47de-bc42-14d163a6277d",
        "name_fi": "Palveluseteli Varhaiskasvatus 85 tuntia / kuukausi",
      },
    },
    {
      "effica": "Vuorohoito 121-140 h/kk",
      "evaka": {
        "default_option": false,
        "id": "9eb82822-82ef-46fb-a123-29cf0af757b7",
        "name_fi": "Palveluseteli Varhaiskasvatus 86–139 tuntia / kuukausi",
      },
    },
    {
      "effica": "Vuorohoito 141-155 h/kk",
      "evaka": {
        "default_option": false,
        "id": "f5d32585-2c78-4434-95d6-30b446db7d4d",
        "name_fi": "Palveluseteli Varhaiskasvatus yli 140 tuntia / kuukausi",
      },
    },
    {
      "effica": "Vuorohoito yli 171 h/kk",
      "evaka": {
        "default_option": false,
        "id": "f5d32585-2c78-4434-95d6-30b446db7d4d",
        "name_fi": "Palveluseteli Varhaiskasvatus yli 140 tuntia / kuukausi",
      },
    },
    {
      "effica": "Pph/ aikap. 141-155 h/kk",
      "evaka": {
        "default_option": false,
        "id": "f5d32585-2c78-4434-95d6-30b446db7d4d",
        "name_fi": "Palveluseteli Varhaiskasvatus yli 140 tuntia / kuukausi",
      },
    },
    {
      "effica": "Pph/ aikap. 156-170 h/kk",
      "evaka": {
        "default_option": false,
        "id": "f5d32585-2c78-4434-95d6-30b446db7d4d",
        "name_fi": "Palveluseteli Varhaiskasvatus yli 140 tuntia / kuukausi",
      },
    },
    {
      "effica": "Pph/ aikap. yli 171 h/kk",
      "evaka": {
        "default_option": false,
        "id": "f5d32585-2c78-4434-95d6-30b446db7d4d",
        "name_fi": "Palveluseteli Varhaiskasvatus yli 140 tuntia / kuukausi",
      },
    },
    {
      "effica": "Pkoti/ aikap. 86-120 h/kk",
      "evaka": {
        "default_option": false,
        "id": "9eb82822-82ef-46fb-a123-29cf0af757b7",
        "name_fi": "Palveluseteli Varhaiskasvatus 86–139 tuntia / kuukausi",
      },
    },
    {
      "effica": "*Kokopäivähoito 12 pv/kk",
      "evaka": {
        "default_option": false,
        "id": "36c8c2ed-7543-47de-bc42-14d163a6277d",
        "name_fi": "Palveluseteli Varhaiskasvatus 85 tuntia / kuukausi",
      },
    },
    {
      "effica": "Pkoti/ aikap. 121-140 h/kk",
      "evaka": {
        "default_option": false,
        "id": "9eb82822-82ef-46fb-a123-29cf0af757b7",
        "name_fi": "Palveluseteli Varhaiskasvatus 86–139 tuntia / kuukausi",
      },
    },
    {
      "effica": "Pkoti/ aikap. 141-155 h/kk",
      "evaka": {
        "default_option": false,
        "id": "f5d32585-2c78-4434-95d6-30b446db7d4d",
        "name_fi": "Palveluseteli Varhaiskasvatus yli 140 tuntia / kuukausi",
      },
    },
    {
      "effica": "Pkoti/ aikap. 156-170 h/kk",
      "evaka": {
        "default_option": false,
        "id": "f5d32585-2c78-4434-95d6-30b446db7d4d",
        "name_fi": "Palveluseteli Varhaiskasvatus yli 140 tuntia / kuukausi",
      },
    },
    {
      "effica": "Pkoti/ aikap. yli 171 h/kk",
      "evaka": {
        "default_option": false,
        "id": "f5d32585-2c78-4434-95d6-30b446db7d4d",
        "name_fi": "Palveluseteli Varhaiskasvatus yli 140 tuntia / kuukausi",
      },
    },
  ],
}
`;

exports[`hameenkyro placement preschool 1`] = `
[
  {
    "effica_ssn": "010124A920A",
    "placement_end_date": "2024-05-31",
    "placement_place_guarantee": false,
    "placement_start_date": "2023-08-09",
    "placement_type": "PRESCHOOL",
    "placement_unit_id": "92b6e496-d273-11ee-9265-db72361ab90d",
    "service_need_end_date": "2024-05-31",
    "service_need_option_id": null,
    "service_need_shift_care": "NONE",
    "service_need_start_date": "2023-08-09",
  },
]
`;

exports[`hameenkyro placement preschool 2`] = `
[
  {
    "evaka_ssn": "010124A920A",
    "placement_end_date": "2024-05-31",
    "placement_place_guarantee": false,
    "placement_start_date": "2023-08-09",
    "placement_type": "PRESCHOOL",
    "placement_unit": "92b6e496-d273-11ee-9265-db72361ab90d",
    "service_need_end_date": null,
    "service_need_option_id": null,
    "service_need_shift_care": null,
    "service_need_start_date": null,
  },
]
`;

exports[`hameenkyro placement preschool daycare 1`] = `
[
  {
    "effica_ssn": "010124A920A",
    "placement_end_date": "2023-08-08",
    "placement_place_guarantee": false,
    "placement_start_date": "2023-08-01",
    "placement_type": "PRESCHOOL_DAYCARE_ONLY",
    "placement_unit_id": "92b6e496-d273-11ee-9265-db72361ab90d",
    "service_need_end_date": "2023-08-08",
    "service_need_option_id": "0a58d934-6fd1-11ed-a75e-c353faef5858",
    "service_need_shift_care": "NONE",
    "service_need_start_date": "2023-08-02",
  },
  {
    "effica_ssn": "010124A920A",
    "placement_end_date": "2024-05-31",
    "placement_place_guarantee": false,
    "placement_start_date": "2023-08-09",
    "placement_type": "PRESCHOOL_DAYCARE",
    "placement_unit_id": "92b6e496-d273-11ee-9265-db72361ab90d",
    "service_need_end_date": "2024-05-31",
    "service_need_option_id": "1c5d7ea4-669f-4b4e-8593-353be4c9cea0",
    "service_need_shift_care": "NONE",
    "service_need_start_date": "2023-08-09",
  },
  {
    "effica_ssn": "010124A920A",
    "placement_end_date": "2024-07-31",
    "placement_place_guarantee": false,
    "placement_start_date": "2024-06-01",
    "placement_type": "PRESCHOOL_DAYCARE_ONLY",
    "placement_unit_id": "92b6e496-d273-11ee-9265-db72361ab90d",
    "service_need_end_date": "2024-07-31",
    "service_need_option_id": "0a58d934-6fd1-11ed-a75e-c353faef5858",
    "service_need_shift_care": "NONE",
    "service_need_start_date": "2024-06-01",
  },
]
`;

exports[`hameenkyro placement preschool daycare 2`] = `
[
  {
    "evaka_ssn": "010124A920A",
    "placement_end_date": "2023-08-08",
    "placement_place_guarantee": false,
    "placement_start_date": "2023-08-01",
    "placement_type": "PRESCHOOL_DAYCARE_ONLY",
    "placement_unit": "92b6e496-d273-11ee-9265-db72361ab90d",
    "service_need_end_date": "2023-08-01",
    "service_need_option_id": "0a58d934-6fd1-11ed-a75e-c353faef5858",
    "service_need_shift_care": "NONE",
    "service_need_start_date": "2023-08-01",
  },
  {
    "evaka_ssn": "010124A920A",
    "placement_end_date": "2023-08-08",
    "placement_place_guarantee": false,
    "placement_start_date": "2023-08-01",
    "placement_type": "PRESCHOOL_DAYCARE_ONLY",
    "placement_unit": "92b6e496-d273-11ee-9265-db72361ab90d",
    "service_need_end_date": "2023-08-08",
    "service_need_option_id": "0a58d934-6fd1-11ed-a75e-c353faef5858",
    "service_need_shift_care": "NONE",
    "service_need_start_date": "2023-08-02",
  },
  {
    "evaka_ssn": "010124A920A",
    "placement_end_date": "2024-05-31",
    "placement_place_guarantee": false,
    "placement_start_date": "2023-08-09",
    "placement_type": "PRESCHOOL_DAYCARE",
    "placement_unit": "92b6e496-d273-11ee-9265-db72361ab90d",
    "service_need_end_date": "2024-05-31",
    "service_need_option_id": "1c5d7ea4-669f-4b4e-8593-353be4c9cea0",
    "service_need_shift_care": "NONE",
    "service_need_start_date": "2023-08-09",
  },
  {
    "evaka_ssn": "010124A920A",
    "placement_end_date": "2024-07-31",
    "placement_place_guarantee": false,
    "placement_start_date": "2024-06-01",
    "placement_type": "PRESCHOOL_DAYCARE_ONLY",
    "placement_unit": "92b6e496-d273-11ee-9265-db72361ab90d",
    "service_need_end_date": "2024-07-31",
    "service_need_option_id": "0a58d934-6fd1-11ed-a75e-c353faef5858",
    "service_need_shift_care": "NONE",
    "service_need_start_date": "2024-06-01",
  },
]
`;

exports[`hameenkyro placement private voucher service 1`] = `
[
  {
    "effica_ssn": "010124A920A",
    "placement_end_date": "2024-05-31",
    "placement_place_guarantee": false,
    "placement_start_date": "2023-08-09",
    "placement_type": "DAYCARE",
    "placement_unit_id": "ccf269c5-8677-452c-b2b7-e476a031f7b0",
    "service_need_end_date": "2024-05-31",
    "service_need_option_id": "36c8c2ed-7543-47de-bc42-14d163a6277d",
    "service_need_shift_care": "NONE",
    "service_need_start_date": "2023-08-09",
  },
]
`;

exports[`hameenkyro placement private voucher service 2`] = `
[
  {
    "evaka_ssn": "010124A920A",
    "placement_end_date": "2024-05-31",
    "placement_place_guarantee": false,
    "placement_start_date": "2023-08-09",
    "placement_type": "DAYCARE",
    "placement_unit": "ccf269c5-8677-452c-b2b7-e476a031f7b0",
    "service_need_end_date": "2024-05-31",
    "service_need_option_id": "36c8c2ed-7543-47de-bc42-14d163a6277d",
    "service_need_shift_care": "NONE",
    "service_need_start_date": "2023-08-09",
  },
]
`;
